homeassistant/components/sma/sensor.py

```diff
@@ -84,7 +84,13 @@ def __init__(
     @property
     def name(self) -> str:
         """Return the name of the sensor."""
-        return self._sensor.name
+        name_prefix = (
+            self._attr_device_info.get("name", "SMA")
+            if self._attr_device_info is not None
+            else "SMA"
+        )
```

Comment:
We generally avoid multiline ternary

Alternatively: 

```suggestion
		name_prefix =  "SMA"
		if self._attr_device_info is not None:
			name_prefix = self._attr_device_info.get("name", "SMA")
```

Reply:
Looks much cleaner! Thanks!